### DISCUSSION TITLE
Does a hylang-flavored version of xonsh exist?

### DISCUSSION THREAD

**Discussion by gtusr** [xonsh/xonsh#5184] [General]
If not, how difficult would this be to implement?

I'm looking for something similar to emacs' eshell but with python-oriented semantics.

**Comment by anki-code** [comment]
Take a look into macro calls to invoke any language:
* https://xon.sh/tutorial_macros.html#context-manager-macros
* https://github.com/anki-code/xontrib-sh
* https://github.com/anki-code/xontrib-macro
